cloudformation/slack-router-cfn-template.js

```diff
@@ -0,0 +1,119 @@
+const cf = require('@mapbox/cloudfriend')
+
+const Parameters = {
+  BucketName: {
+    Type: 'String',
+    Description: 'Name of S3 bucket where Lambda code is saved',
+    Default: 'stork-us-east-1',
+  },
+  GitSha: {
+    Type: 'String',
+    Description: 'Git SHA of latest commit for Lambda function',
+  },
+}
+
+const Resources = {
+  SlackRouterApi: {
+    Type: 'AWS::ApiGatewayV2::Api',
+    Properties: {
+      Name: 'slack-router-api',
+      ProtocolType: 'HTTP',
+    },
+  },
+  SlackToLambdaIntegration: {
+    Type: 'AWS::ApiGatewayV2::Integration',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+      Description: 'Integration for Slack Slash Commands',
+      IntegrationType: 'AWS_PROXY',
+      IntegrationUri: cf.join('', [
+        'arn:',
+        cf.partition,
+        ':apigateway:',
+        cf.region,
+        ':lambda:path/2015-03-31/functions/',
+        cf.getAtt('SlackRouterLambda', 'Arn'),
+        '/invocations',
+      ]),
+      PayloadFormatVersion: '2.0',
+    },
+  },
+  SlackToLambdaRoute: {
+    Type: 'AWS::ApiGatewayV2::Route',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+      AuthorizationType: 'NONE',
+      RouteKey: 'POST /slack-to-lambda',
+      Target: cf.join('/', [
+        'integrations',
+        cf.ref('SlackToLambdaIntegration'),
+      ]),
+    },
+  },
+  SlackToLambdaDeploy: {
+    Type: 'AWS::ApiGatewayV2::Deployment',
+    DependsOn: 'SlackToLambdaRoute',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+    },
+  },
+  SlackRouterApiProdStage: {
+    Type: 'AWS::ApiGatewayV2::Stage',
+    Properties: {
+      ApiId: cf.ref('SlackRouterApi'),
+      AutoDeploy: 'true',
+      DeploymentId: cf.ref('SlackToLambdaDeploy'),
+      StageName: 'prod',
+    },
+  },
+  SlackRouterPermission: {
+    Type: 'AWS::Lambda::Permission',
+    Properties: {
+      Action: 'lambda:InvokeFunction',
+      FunctionName: cf.ref('SlackRouterLambda'),
+      Principal: 'apigateway.amazonaws.com',
+      SourceArn: cf.arn(
+        'execute-api',
+        cf.join('', [cf.ref('SlackRouterApi'), '/*/*'])
+      ),
+    },
+  },
+}
+
+const lambda = new cf.shortcuts.Lambda({
+  LogicalName: 'SlackRouterLambda',
+  Handler: 'cloudformation/slack-router-lambda.handler',
```

Comment:
As far as I know, the handler _has_ to be named, `index.js` file in the root directory of the zip file. In case you notice errors around "handler not found", try this as the first means to debug.

Reply:
This works :) Since the Stork automated code deployment zips the whole repo the handler needs to be accessed through a path since it's not in the root directory. You can assign the handler any name as well (index is the default however).